src/dmd/expressionsem.d

```diff
@@ -3238,6 +3238,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         result = e;
     }
 
+    override void visit(CompoundLiteralExp cle)
+    {
+        static if (LOGSEMANTIC)
+        {
+            printf("CompoundLiteralExp::semantic('%s')\n", cle.toChars());
+        }
+        Type t = cle.type.typeSemantic(cle.loc, sc);
+        auto init = initializerSemantic(cle.initializer, sc, t, INITnointerpret);
+        auto e = initializerToExpression(init, t);
+        if (!e)
+        {
+            error(cle.loc, "cannot convert initializer `%s` to expression", init.toChars());
```

Comment:
This error is not covered by tests.

Reply:
Earlier, I mentioned that I was not going to do coverage for error messages, due to the fluidity and changes going on. Instead, I concentrate on getting correct examples to work, as those shouldn't change.